src/low-level/imap/mailimap_parser.c

```diff
@@ -6017,14 +6017,16 @@ static int mailimap_literal_parse_progress(mailstream * fd, MMAPString * buffer,
   }
   
   r = mailimap_crlf_parse(fd, buffer, parser_ctx, &cur_token);
-	if (r == MAILIMAP_ERROR_PARSE) {
-		/* workaround for Lotus Domino IMAP server */
-		mailimap_space_parse(fd, buffer, &cur_token);
-		mailimap_space_parse(fd, buffer, &cur_token);
-	}
-	else if (r != MAILIMAP_NO_ERROR) {
-    res = r;
-    goto err;
+  if (r != MAILIMAP_NO_ERROR) {
```

Comment:
Could you describe what you are fixing here?
The original code seemed fine.

Reply:
The alert is the part expression 'r != MAILIMAP_NO_ERROR' is always false.
It seems a little confused, as it is.
I believe it was more readable.